modules/derivation/shared/src/main/scala/io/circe/derivation/Discriminator.scala

```diff
@@ -0,0 +1,12 @@
+package io.circe.derivation
+
+//Used to manage different ways of ADT serialization
+
+sealed trait Discriminator
+
+object Discriminator {
+  final case class Embedded(fieldName: String) extends Discriminator
```

Comment:
In circe-generic-extras we tend to use "type field" and "object wrapper" to refer to these approaches. I don't feel too strongly about this, but I think we should standardize one way or the other.

Reply:
Let's me check and apply the same approach